### DISCUSSION TITLE
Commutative Diagrams with diagonal arrows

### DISCUSSION THREAD

**Discussion by Falcelmo** [KaTeX/KaTeX#3632] [General]
Recently I have been trying to find a nice way of creating commutative diagrams with diagonal arrows since this is not possible using the CD environment. I know that you can alternatively use an array to obtain arrows in diagonal directions, but it just doesn't look nearly as nice and you can't add labels to the arrows. For example here are two versions of the same diagram that I have made the first one was made in katex with an array and the second was made in latex using the tikzcd package.
![diagram1](https://user-images.githubusercontent.com/53348058/168503548-cbadcb51-b0b5-4827-89a9-0462708b9d17.png)
![diagram2](https://user-images.githubusercontent.com/53348058/168503550-a390f5ce-c4c1-475c-a952-8d313a6c8399.png)
These pictures came out a bit blurry and I am not sure how to make it look more clear, but it still captures how the length of the arrows makes a big difference in how nice the diagrams look.

Does anyone know any good ways of creating nicer looking diagonal arrows for commutative diagrams in katex other than just pasting images of diagrams from other sources?

**Comment by edemaine** [comment]
You could check out [Quiver](https://github.com/varkor/quiver) which uses a mixture of SVG and KaTeX. Otherwise, I'd probably use the `array` approach but with some manual spacing (e.g. `\!` for negative spaces) to make it look better.
